speech/speech_synthesis_result.go

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
+
+package speech
+
+import (
+	"unsafe"
+
+	"github.com/Microsoft/cognitive-services-speech-sdk-go/common"
+)
+
+// #include <stdlib.h>
+// #include <stdint.h>
+// #include <speechapi_c_common.h>
+// #include <speechapi_c_result.h>
+// #include <speechapi_c_synthesizer.h>
+//
+import "C"
+
+// SpeechSynthesisResult contains detailed information about result of a synthesis operation.
+type SpeechSynthesisResult struct {
+	handle C.SPXHANDLE
+
+	// ResultID specifies the result identifier.
+	ResultID string
+
+	// Reason specifies status of speech synthesis result.
+	Reason common.ResultReason
+
+	// AudioData presents the synthesized audio.
+	AudioData []byte
+
+	// Collection of additional synthesisResult properties.
+	Properties common.PropertyCollection
+}
+
+// Close releases the underlying resources
+func (result SpeechSynthesisResult) Close() {
+	C.synthesizer_result_handle_release(result.handle)
+}
+
+// NewSpeechSynthesisResultFromHandle creates a SpeechSynthesisResult from a handle (for internal use)
+func NewSpeechSynthesisResultFromHandle(handle common.SPXHandle) (*SpeechSynthesisResult, error) {
+	buffer := C.malloc(C.sizeof_char * 1024)
+	defer C.free(unsafe.Pointer(buffer))
+	result := new(SpeechSynthesisResult)
+	result.handle = uintptr2handle(handle)
+	/* ResultId */
+	ret := uintptr(C.synth_result_get_result_id(result.handle, (*C.char)(buffer), 1024))
+	if ret != C.SPX_NOERROR {
+		return nil, common.NewCarbonError(ret)
+	}
+	result.ResultID = C.GoString((*C.char)(buffer))
+	/* Reason */
+	var cReason C.Result_Reason
+	ret = uintptr(C.synth_result_get_reason(result.handle, &cReason))
+	if ret != C.SPX_NOERROR {
+		return nil, common.NewCarbonError(ret)
+	}
+	result.Reason = (common.ResultReason)(cReason)
+	/* AudioData */
+	var cAudioLength C.uint32_t
+	ret = uintptr(C.synth_result_get_audio_length(result.handle, &cAudioLength))
+	if ret != C.SPX_NOERROR {
+		return nil, common.NewCarbonError(ret)
+	}
+	cBuffer := C.malloc(C.sizeof_char * (C.size_t)(cAudioLength))
```

Comment:
Maybe you could use a single buffer for all of this (allocate with size max(cAudioLength, 1024)

Reply:
thx, but seems I cannot use a single buffer here. `synth_result_get_audio_data` is designed to get all audio data ( > 1024 most times) at once

Reply:
yeah, what I meant was to just allocate one buffer with the bigger size of the two

Reply:
thanks, I updated to use one buffer